file-generating/ronn/ronn.def

```diff
@@ -0,0 +1,326 @@
+    echo "No file specified"
+    return 1
+fi
+if [ ! -e "$MD_FILE" ]; then
+    echo "FATAL ERROR: somthing VERY bad happend..."
+    echo "FATAL: variables failed to match"
+    echo "FATAL: 255"
+    echo "$MD_FILE: No such file or directory"
+    return 255
+fi
+
+# check if file is a directory
+if [ -d "$MD_FILE" ]; then
+    echo "$MD_FILE: Is a directory"
+    return 2
+fi
+
+# get the file name without the extention
+if [ "${MD_FILE: -3}" = ".md" ]; then
+    FILE=` echo "$MD_FILE" | rev | cut -c4- | rev `
+elif [ "${MD_FILE: -5}" = ".ronn" ]; then
+    FILE=` echo "$MD_FILE" | rev | cut -c6- | rev `
+else
+    echo "${MD_FILE}: Unknown file extention"
+    return 2
+fi
+
+# check if the output file exist
+if [ "$IS_OUT" = "true" ]; then
+    if [ -e "$OUT_FILE" ]; then
+        echo "${OUT_FILE}: Already exist"
+        return 1
+    fi
+fi
+
+# the tmp file
+TMP_FILE="${FILE}.tmp.1.ronn"
+
+# check is TMP_FILE exist
+if [ -e "$TMP_FILE" ]; then
+    echo "${TMP_FILE}: Already exist"
+    return 1
+fi
+
+# make the TMP_FILE
+cat "$MD_FILE" > "$TMP_FILE"
+
+# check if the manpage exist
+if [ -e "${FILE}.1" ]; then
+    echo "${FILE}.1: Aleady exist"
+    echo "Aborting"
+    rm -f "$TMP_FILE"
+    return 1
+fi
+
+# check if the HTML file exist
+if [ -e "${FILE}.1.html" ]; then
+    echo "${FILE}.1.html: Already exist"
+    echo "Aborting"
+    rm -f "$TMP_FILE"
+    return 1
+fi
+
+# generate the manpage/html
+echo "gererating the manPage..."
+ronn "$TMP_FILE" >/dev/null 2>&1
+
+# rename the output files
+FILE_NAME=` echo "${TMP_FILE%%.*}" `
+
+if [ ! -e "$FILE_NAME.1" ]; then
+    echo "FATAL: unable to get output file from ronn"
+    echo "FATAL: 255"
+    rm -f "$TMP_FILE"
+    exit 255
+fi
+
+if [ "$IS_OUT" = "true" ]; then
+    mv "${FILE_NAME}.1" "$OUT_FILE"
+    mv "${FILE_NAME}.1.html" "${OUT_FILE}.html"
+else
+    mv "${FILE_NAME}.1" "${FILE}.1"
+    mv "${FILE_NAME}.1.html" "${FILE}.1.html"
+fi
+
+# check if gzip file exist
+if [ -e "${FILE}.1.gz" ]; then
+    echo "${FILE}.1.gz: Aleady exist"
+    echo "Not creating gzip file"
+    rm -f "$TMP_FILE"
+    return 1
+fi
+
+# check and gzip the file
+if [ "$IS_OUT" = "true" ]; then
+    if [ -e "${FILE_OUT}.gz" ]; then
+        echo "${FILE_OUT}.gz: Aleady exist"
+        echo "Not creating gzip file"
+        rm -f "$TMP_FILE"
+        return 1
+    fi
+    gzip -k "${OUT_FILE}"
+else
+    gzip -k "${FILE}.1"
+fi
+
+# remove the tmp file
+rm -f "$TMP_FILE"
+
+echo "Done."
+echo
+
+if [ "$IS_OUT" = "true" ]; then
+    FILE="$OUT_FILE"
+else
+    FILE="$FILE.1"
+fi
+
+# more argument prosesing
+if [ "$KEEP_ALL" = "true" ]; then
+    echo "manPage:     ${FILE}"
+    echo "gzip file:   ${FILE}.gz"
+    echo "html file:   ${FILE}.html"
+    return 0
+fi
+if [ "$ONLY_MAN" != "true" ]; then
+    rm -f "${FILE}"
+fi
+if [ "$ONLY_GZIP" != "true" ]; then
+    rm -f "${FILE}.gz"
+fi
+if [ "$ONLY_HTML" != "true" ]; then
+    rm -f "${FILE}.html"
+fi
+
+# what file exist for output message
```

Comment:
maybe add a verb in here: "**verify** which files exist for output message" for example